public/gatk-utils/src/main/java/org/broadinstitute/gatk/utils/MannWhitneyU.java

```diff
@@ -208,9 +217,16 @@ public void add(Number n, USet set) {
      */
     @Requires({"m > 0","n > 0"})
     @Ensures({"! Double.isNaN(result)", "! Double.isInfinite(result)"})
-    private static double getZApprox(int n, int m, long u) {
+    private static double getZApprox(int n, int m, double u, Map<Integer, Integer> tieStructure) {
         double mean = ( ((long)m)*n+1.0)/2;
         double var = (((long) n)*m*(n+m+1.0))/12;
+        if (tieStructure != null) {
+            double varAdjust = 0;
+            for (Map.Entry<Integer, Integer> tie : tieStructure.entrySet()) {
+                varAdjust += tie.getValue() * (Math.pow(tie.getKey(), 3) - tie.getKey());
+            }
+            var -= varAdjust / ((n + m) * (n + m - 1));
```

Comment:
Your adjustment is off here - you need

var -= varAdjust * (n * m) / (12 * (n + m) * (n + m - 1))